### PR TITLE
log_backup: fix stale task when version changes

### LINKED ISSUE #16469: pitr checkpoint ts lag reached more than 8h after inject network partition between one of tikv and pd leader
### What version of TiKV are you using?
./tikv-server -V
 TiKV 
Release Version:   8.0.0-alpha
Edition:           Community
Git Commit Hash:   43d0e061159100092d89147480a94b51b6a158b4
Git Commit Branch: heads/refs/tags/v8.0.0-alpha
UTC Build Time:    2024-01-26 11:47:08
Rust Version:      rustc 1.77.0-nightly (89e2160c4 2023-12-27)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure trace-async-tasks openssl-vendored
Profile:           dist_release

### What operating system and CPU are you using?
8c/32g

### Steps to reproduce
1、start pitr
2、run workload
go-tpc tpcc run -D tpcc20000 --host tc-tidb.endless-ha-test-oltp-pitr-tps-6570032-1-702 -P4000 --warehouses 20000 -T 32 --ignore-error '2013,1213,1105,1205,8022,8028,9004,9007,1062' --user root --password '' --interval '10s'
3、inject network partition between one of tikv and pd leader

image

### What did you expect?
pitr checkpoint ts lag less than 10mins after fault recover

### What did happened?
pitr checkpoint ts lag reached more than 8h after inject network partition between one of tikv and pd leader
image

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: Close #16469

What's Changed:

```commit-message
Now, `Start` will always put a phantom record in subscription tracer if there isn't one.
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
Fixed a bug that may cause checkpoint not get advanced when a region get split or merged immediately after becoming leader.
```

### CODE REVIEW

File: components/backup-stream/src/subscription_manager.rs [hunk 1]
```diff
@@ -1115,7 +1122,19 @@ mod test {
                 .unwrap()
         }
 
+        fn insert_and_start_region(&self, region: Region) {
+            self.insert_region(region.clone());
```
Comment (Leavrth) [inline components/backup-stream/src/subscription_manager.rs:1126]: duplicated `region.clone()` in line 1141

**Review by Leavrth — APPROVED** [review]

File: components/backup-stream/src/subscription_manager.rs [hunk 2]
```diff
@@ -543,6 +543,7 @@ where
         let tx = tx.unwrap();
         // tikv_util::Instant cannot be converted to std::time::Instant :(
         let start = std::time::Instant::now();
+        info!("Scheduing subscription."; utils::slog_region(&region), "after" => ?backoff, "handle" => ?handle);
```
Comment (overvenus) [inline components/backup-stream/src/subscription_manager.rs:546]: Maybe too verbose? I find there lots of logs from PiTR in tikv.log.

**Review by overvenus — COMMENTED** [review]
rest lgtm

File: components/backup-stream/src/subscription_manager.rs (hunk 2 above)
Comment (YuJuncen) [reply components/backup-stream/src/subscription_manager.rs:546]: Generally this should be print once retrying. But I guess with the async-backtrace we can make it debug level.

File: components/backup-stream/tests/failpoints/mod.rs [hunk 3]
```diff
@@ -268,6 +268,7 @@ mod all {
         fail::cfg("try_start_observe", "2*return").unwrap();
         fail::cfg("try_start_observe0", "off").unwrap();
 
+        std::thread::sleep(Duration::from_secs(2));
```
Comment (overvenus) [inline components/backup-stream/tests/failpoints/mod.rs:271]: Please add some comments

**Review by overvenus — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

**Comment by YuJuncen** [comment]
(It seems the case `failed_during_refresh_region` not pretty stable...)

**Pull request merged**